sushy_oem_idrac/utils.py

```diff
@@ -38,3 +43,32 @@ def reboot_system(system):
         system.refresh()
 
     LOG.info('System powered ON')
+
+@retrying.retry(
+    retry_on_exception=lambda exception: isinstance(exception, Exception),
+    stop_max_attempt_number=_IDRAC_IS_READY_RETRIES,
+    wait_fixed=_IDRAC_IS_READY_RETRY_DELAY_SEC * 1000)
+def wait_for_idrac_ready(oem_manager):
+    is_ready_response = is_idrac_ready(oem_manager)
+    if "LCStatus" in is_ready_response:
+         LOG.info("idrac for node is ready")
+         return True
+    else:
+        err_msg = ('idrac for node is not ready,'
+                'Failed to perform drac operation, Retrying it' )
+        raise sushy.exception.ConnectionError(error=err_msg)
+
+def is_idrac_ready(oem_manager):
+
+    try:
+        response = asynchronous.http_call(
+                            oem_manager._conn, 'post',
+                            oem_manager.get_remote_service_api_uri,
+                            headers=oem_manager.HEADERS,
+                            data={},
+                            verify=False)
+        data = response.json()
+    except Exception as err:
```

Comment:
I think we should catch sushy.exceptions.ServerSideError and sushy.exceptions.BadRequestError here, log an error, and reraise the caught exception.